src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperDesignTimeDescriptorFactory.cs

```diff
@@ -17,8 +18,11 @@ namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
     /// Factory for providing <see cref="TagHelperDesignTimeDescriptor"/>s from <see cref="Type"/>s and
     /// <see cref="TagHelperAttributeDesignTimeDescriptor"/>s from <see cref="PropertyInfo"/>s.
     /// </summary>
-    public static class TagHelperDesignTimeDescriptorFactory
+    public class TagHelperDesignTimeDescriptorFactory
     {
+        private readonly Dictionary<int, XmlDocumentationProvider> _documentationProviderCache =
+            new Dictionary<int, XmlDocumentationProvider>();
```

Comment:
Hmmm, if the same Razor parser is invoked on two separate threads simultaneously then possibly. It ultimately depends on how the user is setting the `TagHelperDescriptorResolver`, by default it's set once per `MvcRazorHost`. Going to swap to `ConcurrentDictionary`.